### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by spotbot2k**
Adds docs about the context guard and bumps the version in the usage example. If you would agree I can continue to contribute to the docs and add further documents.

### CODE REVIEW

**Comment by solverat** [comment]
Hey @spotbot2k. I'm sorry for the late response, but we had to push things forward to finally support pimcore x. This moment has finally arrived, and it would be great if you could rebase your PR to the latest commit on master branch. 

Since we're still a little behind in documentation (although there are some more within v2 :)) it would be **super awesome** if you're willing to contribute even more to our documentation base.

**Comment by spotbot2k** [comment]
@solverat Hi, sorry, I've been away on holiday for a while. I'll try to update the branch ASAP, lots to do in january as usual.

**Comment by solverat** [comment]
@spotbot2k: No hurries, I know the pain of January! :)

File: docs/200_ContextGuard.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+The context guard add an additional layer of protection and controll ontop of the set of your indexing rules. It allows you to filter the result and avoid indexing unvanted content.
+
+To  implement a guard add a service that would implement `ContextGuardInterface`.
+
+## Example Setup
+
+``` yaml
+services:
+  AppBundle\DynamicSearch\Guard\DefaultContextGuard:
```
Comment (solverat) [inline docs/200_ContextGuard.md:11]: Please use `App` Namespace here (Symfony 5 Standard)

File: docs/200_ContextGuard.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    tags:
+      - { name: dynamic_search.context_guard }
+```
+This simple guard will prevent the document with the id 1 (it is the root document in the pimcore document tree) to be added to the index.
+
+``` php
+<?php
+
+namespace AppBundle\DynamicSearch\Guard;
```
Comment (solverat) [inline docs/200_ContextGuard.md:20]: Please use `App` Namespace here (Symfony 5 Standard)

Comment (spotbot2k) [reply docs/200_ContextGuard.md:20]: Thx, I will check al lthe steps again on a fresh setup and add some updates

File: docs/200_ContextGuard.md (hunk 1 above)
Comment (spotbot2k) [reply docs/200_ContextGuard.md]: I've updated the namespace and the path (to `config/config` in v5). Some more changes have been done to proovide a working setup based on the official demo.

**Comment by spotbot2k** [comment]
Sorry, have forgotten to set to draft. The basic example is now ready. It allows to setup Trinity+Lucene combination with a simple index into the Pimcore X demo and shows how to test the index.

**Comment by solverat** [comment]
@spotbot2k perfect! Thank you!

**Review by solverat — APPROVED** [review]

**Pull request merged**